### PR TITLE
adding gray preview for viewing lighting changes directly

### PR DESCRIPTION
**Pull request by kotoole1**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/lukemetz/Paranormal/pull/104%23issuecomment-88530508%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23discussion_r27779541%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23discussion_r27779544%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23discussion_r27779557%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23issuecomment-89831745%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23issuecomment-88530508%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Checked%20out%20branch%20and%20after%20a%20clean%20build%2C%20branch%20produces%20the%20following%20cryptic%20error%3A%5Cr%5CnCommand%20/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc%20failed%20with%20exit%20code%201%22%2C%20%22created_at%22%3A%20%222015-04-01T15%3A51%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1645633%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/NCEghtebas%22%7D%7D%2C%20%7B%22body%22%3A%20%22Do%20the%20stuff%2C%20then%20%3Aship%3A%22%2C%20%22created_at%22%3A%20%222015-04-05T18%3A52%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20382a611328b1102ab417cf0a7218399831b19143%20Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23discussion_r27779541%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Left%20in%20NSImageHelper%22%2C%20%22created_at%22%3A%20%222015-04-05T18%3A50%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift%3AL107-127%22%7D%2C%20%22Pull%20382a611328b1102ab417cf0a7218399831b19143%20Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift%2028%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23discussion_r27779544%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20would%20just%20do%20self.previewSprite%20%3D%20CCSprite%20.....%22%2C%20%22created_at%22%3A%20%222015-04-05T18%3A50%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift%3AL130-139%22%7D%2C%20%22Pull%20382a611328b1102ab417cf0a7218399831b19143%20Paranormal/Paranormal/Panels/Preview/PreviewSpriteUtils.swift%2030%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/104%23discussion_r27779557%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Could%20you%20throw%20a%20comment%20here%20that%20this%20function%20needs%20to%20be%20run%20on%20GPUImage%20thread%3F%20Could%20lead%20to%20miss%20leading%20in%20the%20future.%22%2C%20%22created_at%22%3A%20%222015-04-05T18%3A52%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/Paranormal/Panels/Preview/PreviewSpriteUtils.swift%3AL54-94%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/104#issuecomment-88530508'>General Comment</a></b>
- <a href='https://github.com/NCEghtebas'></a> Checked out branch and after a clean build, branch produces the following cryptic error:
Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc failed with exit code 1
- <a href='https://github.com/lukemetz'></a> Do the stuff, then :ship:
- [ ] <a href='#crh-comment-Pull 382a611328b1102ab417cf0a7218399831b19143 Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift 7'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/104#discussion_r27779541'>File: Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift:L107-127</a></b>
- <a href='https://github.com/lukemetz'></a> Left in NSImageHelper
- [ ] <a href='#crh-comment-Pull 382a611328b1102ab417cf0a7218399831b19143 Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift 28'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/104#discussion_r27779544'>File: Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift:L130-139</a></b>
- <a href='https://github.com/lukemetz'></a> I would just do self.previewSprite = CCSprite .....
- [ ] <a href='#crh-comment-Pull 382a611328b1102ab417cf0a7218399831b19143 Paranormal/Paranormal/Panels/Preview/PreviewSpriteUtils.swift 30'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/104#discussion_r27779557'>File: Paranormal/Paranormal/Panels/Preview/PreviewSpriteUtils.swift:L54-94</a></b>
- <a href='https://github.com/lukemetz'></a> Could you throw a comment here that this function needs to be run on GPUImage thread? Could lead to miss leading in the future.

<a href='https://www.codereviewhub.com/lukemetz/Paranormal/pull/104?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/lukemetz/Paranormal/pull/104?approve=1'></a>&nbsp;<a href='https://github.com/lukemetz/Paranormal/pull/104'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by NCEghtebas** [comment]
Checked out branch and after a clean build, branch produces the following cryptic error:
Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc failed with exit code 1

File: Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift [hunk 1]
```diff
@@ -107,15 +107,20 @@ class PreviewLayer: CCNode {
         }
     }
 
-    func updateBaseImage(image : NSImage) {
+    func updateBaseImage(image : NSImage, keepNormalMap: Bool) {
         ThreadUtils.runCocos { () -> Void in
+            NSImageHelper.writeToFile(image, path: "/users/kotoole/Desktop/baseImage.png")
```
Comment (lukemetz) [inline Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift]: Left in NSImageHelper

File: Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift [hunk 2]
```diff
@@ -125,8 +130,9 @@ class PreviewLayer: CCNode {
                 self.updateNodePositions()
             } else {
                 // New session begun, initialize sprite
-                let sprite = CCSprite(texture: PreviewSpriteUtils.spriteTextureForImage(image))
-                self.previewSprite = sprite
+                let previewSprite = CCSprite(texture:
```
Comment (lukemetz) [inline Paranormal/Paranormal/Panels/Preview/PreviewLayer.swift]: I would just do self.previewSprite = CCSprite .....

File: Paranormal/Paranormal/Panels/Preview/PreviewSpriteUtils.swift [hunk 3]
```diff
@@ -53,4 +54,40 @@ class PreviewSpriteUtils: NSObject {
...
+        let rect = CGRectMake(0, 0, CGFloat(width), CGFloat(height))
+        CGContextFillRect(context, rect)
+        let cgImage = CGBitmapContextCreateImage(context)
+
+        let size = NSSize(width: CGFloat(width), height: CGFloat(height))
+        let grayNSImage = NSImage(CGImage: cgImage, size: size)
+        return grayNSImage
+    }
+
+    class func grayImageWithAlphaSource(source: NSImage, brightness: Float) -> NSImage {
```
Comment (lukemetz) [inline Paranormal/Paranormal/Panels/Preview/PreviewSpriteUtils.swift]: Could you throw a comment here that this function needs to be run on GPUImage thread? Could lead to miss leading in the future.

**Comment by lukemetz** [comment]
Do the stuff, then :ship:

**Pull request merged**